src/image.cpp

```diff
@@ -733,19 +733,15 @@ AccessMode ImageFactory::checkMode(ImageType type, MetadataId metadataId) {
   auto r = Exiv2::find(registry, type);
   if (!r)
     throw Error(ErrorCode::kerUnsupportedImageType, static_cast<int>(type));
-  switch (metadataId) {
-    case mdExif:
-      return r->exifSupport_;
-    case mdIptc:
-      return r->iptcSupport_;
-    case mdXmp:
-      return r->xmpSupport_;
-    case mdComment:
-      return r->commentSupport_;
-    default:
-      return amNone;
-  }
-  return {};
+  if (metadataId == mdExif)
+    return r->exifSupport_;
+  if (metadataId == mdIptc)
```

Comment:
[question] I see why you are making this change (I have been there and done this 😛 ) . However, would it not make more sense to have `else if` statements after the first if  ?

Reply:
Not really. else after return is redundant.